crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -172,17 +181,55 @@ impl<'a> ReflectDerive<'a> {
 
                     reflect_mode = Some(ReflectMode::Value);
                 }
+                Meta::NameValue(pair) if pair.path.is_ident(TYPE_PATH_ATTRIBUTE_NAME) => {
+                    let Lit::Str(lit) = pair.lit else {
+                        return Err(syn::Error::new(
+                            pair.span(),
+                            format_args!("`#[{TYPE_PATH_ATTRIBUTE_NAME} = \"...\"]` must be a string literal"),
+                        ));
+                    };
+
+                    custom_path = Some(syn::parse::Parser::parse_str(
+                        parse_path_no_leading_colon,
+                        &lit.value(),
+                    )?);
+                }
+                Meta::NameValue(pair) if pair.path.is_ident(TYPE_NAME_ATTRIBUTE_NAME) => {
+                    let Lit::Str(lit) = pair.lit else {
+                        return Err(syn::Error::new(
+                            pair.span(),
+                            format_args!("`#[{TYPE_NAME_ATTRIBUTE_NAME} = \"...\"]` must be a string literal"),
+                        ));
+                    };
+
+                    custom_type_name = Some(parse_str(&lit.value())?);
+                }
```

Comment:
Well, we've updated to `syn2`. But again, this can be addressed in a future PR where we reconsider exactly what kind of syntax we want for attributes.